nbgrader/auth/hubauth.py

```diff
@@ -0,0 +1,123 @@
+"""JupyterHub authenticator."""
+import requests
+import os
+import json
+from flask import request, redirect, abort
+from IPython.utils.traitlets import Unicode, Int, List
+from .base import BaseAuth
+from nbgrader.html.formgrade import blueprint
+
+
+class HubAuth(BaseAuth):
+    """Jupyter hub authenticator."""
+
+    graders = List([], config=True, help="List of JupyterHub user names allowed to grade.")
+
+    proxy_address = Unicode(None, config=True, allow_none=True, help="Address of the configurable-http-proxy server.")
+    proxy_port = Int(8001, config=True, help="Port of the configurable-http-proxy server.")
+    hub_address = Unicode(None, config=True, allow_none=True, help="Address of the hub server.")
```

Comment:
These addresses don't need to allow None, they can have 

```python
def _hub_address_default(self):
    return self._ip
```

No need for the init logic below.